### PR TITLE
Support astral Unicode characters in TextElements and StringLiteral

### LINKED ISSUE #174: Add more Unicode planes to regular_char
Let's extend the `regular_char` production to be more permissive of characters from outside of BMP. A good standard to follow is https://www.w3.org/TR/REC-xml/#NT-Char.

https://github.com/projectfluent/fluent/blob/3c7cd301072b43c5ebeb8a2d8e461d2f47e4fef8/syntax/grammar.mjs#L411-L417

### PR DESCRIPTION
**Pull request by stasm**
Fix #174.

### CODE REVIEW

File: spec/fluent.ebnf [hunk 1]
```diff
@@ -94,7 +94,7 @@ quote               ::= "\""
  * Cf. https://www.w3.org/TR/REC-xml/#NT-Char
  * TODO Add characters from other planes: U+10000 to U+10FFFF.
```
Comment (Pike) [inline spec/fluent.ebnf]: No more TOOD

Comment (stasm) [reply spec/fluent.ebnf]: Duh, thanks :)

**Comment by stasm** [comment]
There's the ES spec: https://ecma-international.org/ecma-262/6.0/#sec-literals-string-literals but it's a bit cryptic.

**Comment by stasm** [comment]
Thanks for the review, I'll merge this once I publish 0.7.

**Pull request merged**